### DISCUSSION TITLE
Report all missing dependencies at once

### DISCUSSION THREAD

**Discussion by meleu** [bashly-framework/bashly#580] [Ideas]
First of all: **WOW**! :open_mouth:

As bash-lover for more than 2 decades, I'm amazed by this project! This is a life-changing tool! Thanks for sharing it with us! :heart:

Now the feature idea:

---

When the script has more than one dependency and the user doesn't have any of them installed, it is a frustrating experience to see only one "missing dependency" notification at each time you try to run the script.

It would be nice if the dependency checking could report all missing dependencies in a single run.

I'm sharing here what I usually do (in a hope it can give you some ideas):

```bash
# declare the dependencies in an array
readonly DEPENDENCIES=(dep1 dep2 depN)

# checkDependencies:
# prints the missing dependency
# finishes with failure status when a missing dependency is found
checkDependencies() {
  local dep
  local failure=false

  for dep in "${DEPENDENCIES[@]}"; do
    command -v "$dep" > /dev/null && continue
    echo "[ERROR]: '$dep' is not in your PATH" >&2
    failure=true
  done

  [[ "$failure" == 'false' ]]
}
```

**Comment by DannyBen** [comment]
Hey - very nice to hear that bashly is useful for others, thanks for that.

As for the feature idea - I like it.
In your vision - should this be something that is configurable, or just the new default behavior?

For reference, I believe the change will be in this view:
https://github.com/DannyBen/bashly/blob/master/lib/bashly/views/dependency/filter.gtx

**Reply by meleu** [reply]
A default behavior sounds reasonable to me.

As, at least in a quick thinking, I don't imagine any scenario where the report-only-one-missing-dependency-per-run behavior could be useful.

**Comment by DannyBen** [comment]
Created #582
